src/profile-logic/process-profile.js

```diff
@@ -560,13 +604,30 @@ function _processSamples(geckoSamples: GeckoSampleStruct): SamplesTable {
   };
 }
 
+// Converts a schema array-based extension list to a list of Extension
+// objects with the same information.
+function _processExtensions(extensions: ExtensionMeta): Extension[] {
+  const result: Extension[] = [];
+
+  const { schema, data } = extensions;
+  for (const ext of data) {
+    result.push({
+      id: ext[schema.id],
+      name: ext[schema.name],
+      baseURL: ext[schema.baseURL],
+    });
+  }
+  return result;
+}
+
 /**
  * Convert the given thread into processed form. See docs/gecko-profile-format for more
  * information.
  */
 function _processThread(
   thread: GeckoThread,
-  processProfile: GeckoProfile
+  processProfile: GeckoProfile,
+  mainProcessProfile: GeckoProfile
```

Comment:
Could you pass in the extension information here, rather than passing in the `mainProcessProfile` and re-processing it for each thread? I found this approach to be a bit confusing, and has some redundant processing.